locked-asset/simple-lock-energy/src/unlock_with_penalty.rs

```diff
@@ -197,24 +197,60 @@ pub trait UnlockWithPenaltyModule:
         token_amount * penalty_percentage / MAX_PERCENTAGE as u64
     }
 
-    fn burn_penalty(&self, token_id: TokenIdentifier, fees_amount: &BigUint) {
+    fn burn_penalty(&self, token_id: TokenIdentifier, token_nonce: Nonce, fees_amount: &BigUint) {
         let fees_burn_percentage = self.fees_burn_percentage().get();
         let burn_amount = fees_amount * fees_burn_percentage as u64 / MAX_PERCENTAGE as u64;
         let remaining_amount = fees_amount - &burn_amount;
+        let locked_token_mapper = self.locked_token();
 
         if burn_amount > 0 {
             self.send().esdt_local_burn(&token_id, 0, &burn_amount);
         }
         if remaining_amount > 0 {
-            self.send_fees_to_collector(token_id, remaining_amount);
+            // self.send_fees_to_collector(token_id, remaining_amount);
+            if self.fees_from_penalty_unlocking().is_empty() {
+                self.fees_from_penalty_unlocking()
+                    .set(NonceAmountPair::new(token_nonce, remaining_amount));
+            } else {
+                let existing_nonce_amount_pair = self.fees_from_penalty_unlocking().get();
+                let new_token_attributes: LockedTokenAttributes<Self::Api> =
+                    locked_token_mapper.get_token_attributes(token_nonce);
+                let new_amount = &existing_nonce_amount_pair.amount + &remaining_amount;
+
+                locked_token_mapper.nft_burn(
+                    existing_nonce_amount_pair.nonce,
+                    &existing_nonce_amount_pair.amount,
+                );
+                locked_token_mapper.nft_burn(token_nonce, &remaining_amount);
+
+                let new_token_payment =
+                    locked_token_mapper.nft_create(new_amount, &new_token_attributes);
+                self.fees_from_penalty_unlocking().set(NonceAmountPair::new(
+                    new_token_payment.token_nonce,
+                    new_token_payment.amount,
+                ));
+            }
```

Comment:
Should probably be a separate function.

Also, this is wrong. It should merge the tokens, by merging their attributes. Use `LockedAmountAttributesPair`'s `merge_with` method for this.

Reply:
An example can be seen in token_merging.rs of simple-lock-energy.